modules/Router.js

```diff
@@ -91,7 +90,7 @@ const Router = React.createClass({
       } else {
         // Keep the identity of this.router because of a caveat in ContextUtils:
         // they only work if the object identity is preserved.
-        assign(this.router, this.createRouterObject(state))
+        updateRouterObjectState(this.router, state)
```

Comment:
I wanted to keep the shape of `router` in one place, rather than having coupling to it's shape spread into two places.